### PR TITLE
Add flag to send logs to logger

### PR DESCRIPTION
**Pull request by vermakhushboo**
## What does this PR do?

Added new attribute to Errors to decide whether to log an error in logger (Sentry).

## Test Plan

Tested manually

## Related PRs and Issues

- (Related PR or issue)

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [ ] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

**Review by eldadfux — APPROVED** [review]

File: app/config/errors.php [hunk 1]
```diff
@@ -764,4 +765,12 @@
         'description' => 'An error occurred on the provider\'s side. Please try again later.',
         'code' => 400,
     ],
+
+    /** Health */
+    Exception::QUEUE_SIZE_EXCEEDS => [
```
Comment (christyjacob4) [inline app/config/errors.php:770]:
```suggestion
    Exception::QUEUE_SIZE_EXCEEDED => [
```

File: app/config/errors.php [hunk 2]
```diff
@@ -209,6 +209,7 @@
         'name' => Exception::USER_AUTH_METHOD_UNSUPPORTED,
         'description' => 'The requested authentication method is either disabled or unsupported. Please check the supported authentication methods in the Appwrite console.',
         'code' => 501,
+        'log' => false,
```
Comment (christyjacob4) [inline app/config/errors.php:212]: is this expected here?

File: src/Appwrite/Extend/Exception.php [hunk 3]
```diff
@@ -273,4 +278,14 @@ public function setType(string $type): void
     {
         $this->type = $type;
     }
+
+    /**
+     * Get the log of the exception.
+     *
+     * @return bool
+     */
+    public function getLog(): bool
```
Comment (christyjacob4) [inline src/Appwrite/Extend/Exception.php:287]: This is a misleading name for the getter.

The `log` attribute is a boolean that controls whether a log should be pushed or not... But the name `log` does not indicate this...  we've to use a better name for the attribute and thereby the getter

File: app/controllers/general.php [hunk 4]
```diff
@@ -608,9 +608,14 @@ function router(App $utopia, Database $dbForConsole, SwooleRequest $swooleReques
 
         $version = App::getEnv('_APP_VERSION', 'UNKNOWN');
         $route = $utopia->getRoute();
+        $publishLog = true;
```
Comment (christyjacob4) [inline app/controllers/general.php:611]:
```suggestion
        $publish = true;
```

File: app/controllers/general.php [hunk 5]
```diff
@@ -608,9 +608,14 @@ function router(App $utopia, Database $dbForConsole, SwooleRequest $swooleReques
...
+
+        if ($error instanceof AppwriteException) {
+            $publishLog = $error->getLog();
+        }
 
         if ($logger) {
```
Comment (christyjacob4) [inline app/controllers/general.php:617]:
```suggestion
        if ($logger && publish) {
```

File: app/controllers/general.php [hunk 6]
```diff
@@ -608,9 +608,14 @@ function router(App $utopia, Database $dbForConsole, SwooleRequest $swooleReques
...
-            if ($error->getCode() >= 500 || $error->getCode() === 0) {
+            if (($error->getCode() >= 500 || $error->getCode() === 0) && $publishLog) {
```
Comment (christyjacob4) [inline app/controllers/general.php:618]:
```suggestion
            if ($error->getCode() >= 500 || $error->getCode() === 0) {
```

File: src/Appwrite/Extend/Exception.php (hunk 3 above)
Comment (vermakhushboo) [reply src/Appwrite/Extend/Exception.php:287]: Changing it to `publishLog`

File: app/config/errors.php [hunk 7]
```diff
@@ -764,4 +765,12 @@
         'description' => 'An error occurred on the provider\'s side. Please try again later.',
         'code' => 400,
     ],
+
+    /** Health */
+    Exception::QUEUE_SIZE_EXCEEDED => [
+        'name' => Exception::QUEUE_SIZE_EXCEEDED,
+        'description' => 'Queue size threshold hit.',
+        'code' => 503,
+        'publishLog' => false
```
Comment (christyjacob4) [inline app/config/errors.php:774]:
```suggestion
        'publish' => false
```

File: src/Appwrite/Extend/Exception.php [hunk 8]
```diff
@@ -273,4 +278,14 @@ public function setType(string $type): void
     {
         $this->type = $type;
     }
+
+    /**
+     * Get the value of publishLog for the exception.
+     *
+     * @return bool
+     */
+    public function getPublishLog(): bool
```
Comment (christyjacob4) [inline src/Appwrite/Extend/Exception.php:287]:
```suggestion
    public function isPublishable(): bool
```

File: src/Appwrite/Extend/Exception.php [hunk 9]
```diff
@@ -273,4 +278,14 @@ public function setType(string $type): void
...
+    {
+        return $this->publishLog;
```
Comment (christyjacob4) [inline src/Appwrite/Extend/Exception.php:289]:
```suggestion
        return $this->publish;
```

File: src/Appwrite/Extend/Exception.php [hunk 10]
```diff
@@ -229,12 +229,16 @@ class Exception extends \Exception
...
+    public const REALTIME_MESSAGE_FORMAT_INVALID    = 'realtime_message_format_invalid';
+    public const REALTIME_TOO_MANY_MESSAGES         = 'realtime_too_many_messages';
+    public const REALTIME_POLICY_VIOLATION          = 'realtime_policy_violation';
+
+    /** Health */
+    public const QUEUE_SIZE_EXCEEDED                 = 'queue_size_exceeded';
 
     protected string $type = '';
     protected array $errors = [];
+    protected bool $publishLog = true;
```
Comment (christyjacob4) [inline src/Appwrite/Extend/Exception.php:241]:
```suggestion
    protected bool $publish = true;
```

File: src/Appwrite/Extend/Exception.php [hunk 11]
```diff
@@ -244,6 +248,7 @@ public function __construct(string $type = Exception::GENERAL_UNKNOWN, string $m
         if (isset($this->errors[$type])) {
             $this->code = $this->errors[$type]['code'];
             $this->message = $this->errors[$type]['description'];
+            $this->publishLog = $this->errors[$type]['publishLog'] ?? true;
```
Comment (christyjacob4) [inline src/Appwrite/Extend/Exception.php:251]:
```suggestion
            $this->publish = $this->errors[$type]['publish'] ?? true;
```

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**